Realm/RLMConstants.h

```diff
@@ -18,6 +18,14 @@
 
 #import <Foundation/Foundation.h>
 
+// For compatibility with Xcode 7, before extensible string enums were introduced,
+#ifdef NS_EXTENSIBLE_STRING_ENUM
+#define RLM_EXTENSIBLE_STRING_ENUM_CASE_SWIFT_NAME(_, extensible_string_enum) NS_SWIFT_NAME(extensible_string_enum)
+#else
+#define NS_EXTENSIBLE_STRING_ENUM
```

Comment:
I think you've misunderstood what I'm referring to. By doing `#define NS_EXTENSIBLE_STRING_ENUM` any code that imports our header and does a check of the form `#ifdef NS_EXTENSIBLE_STRING_ENUM` will now see that as defined, but it's not defined in the way that they're likely to expect. What we should do instead is use `RLM_EXTENSIBLE_STRING_ENUM` in our headers in place of `NS_EXTENSIBLE_STRING_ENUM`, and conditionally define it to either `NS_EXTENSIBLE_STRING_ENUM` or nothing.